src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/PowerScoreCalculator.java

```diff
@@ -2,31 +2,66 @@
 
 import de.unistuttgart.iste.gits.common.event.UserProgressLogEvent;
 import de.unistuttgart.iste.gits.generated.dto.Content;
+import de.unistuttgart.iste.gits.generated.dto.RewardChangeReason;
 import de.unistuttgart.iste.gits.reward.persistence.dao.AllRewardScoresEntity;
 import de.unistuttgart.iste.gits.reward.persistence.dao.RewardScoreEntity;
+import de.unistuttgart.iste.gits.reward.persistence.dao.RewardScoreLogEntry;
 import org.springframework.stereotype.Component;
 
 import java.util.List;
 
 @Component
 public class PowerScoreCalculator implements ScoreCalculator {
 
-
+    public static final double HEALTH_FITNESS_MULTIPLIER = 0.1;
     @Override
     public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores, List<Content> contents) {
-        RewardScoreEntity powerScoreBefore = allRewardScores.getPower();
 
-        // TODO calculate power score
+        RewardScoreEntity growth = allRewardScores.getGrowth();
+        RewardScoreEntity strength = allRewardScores.getStrength();
+        RewardScoreEntity health = allRewardScores.getHealth();
+        RewardScoreEntity fitness = allRewardScores.getFitness();
+        RewardScoreEntity power = allRewardScores.getPower();
+
+
+        double powerValueDouble = (growth.getValue() + strength.getValue()) + HEALTH_FITNESS_MULTIPLIER * (health.getValue() + fitness.getValue()) * (growth.getValue() + strength.getValue());
+        int powerValueInt = (int) Math.round(powerValueDouble);
+
+        RewardScoreLogEntry logEntry = RewardScoreLogEntry.builder()
+
```

Comment:
the other fields of the log entry also have to be set! It is not allowed to have "null" in the database! For associated contents you can use an empty list.